paper-input-container.html

```diff
@@ -515,8 +515,7 @@
     _handleValue: function(inputElement) {
       var value = this._inputElementValue;
 
-      // type="number" hack needed because this.value is empty until it's valid
```

Comment:
I updated the JSBin: http://jsbin.com/suruje/edit?html,output. If I type `111eee` I get: 
<img width="165" alt="screen shot 2016-04-13 at 11 45 34 pm" src="https://cloud.githubusercontent.com/assets/1369170/14519146/d4c7ac4c-01d1-11e6-9e1d-f79d71b95678.png">

With your change patched in, if I type `111` everything is fine, but as soon as I start typing the `e`s I get:
<img width="173" alt="screen shot 2016-04-13 at 11 49 25 pm" src="https://cloud.githubusercontent.com/assets/1369170/14519244/747fd58e-01d2-11e6-83e0-473ddc4e7413.png">

This is because the moment you type the first `e`, internally the native `input` goes "nope, invalid", and sets a) the validity to false and b) the `value` property to `""`. This means that the code above won't run (with your change), and it will set `inputHasContent` to false, which will unfloat the label.